eZ/Publish/Core/REST/Server/Controller/User.php

```diff
@@ -179,16 +179,31 @@ public function loadUser($userId)
         $user = $this->userService->loadUser($userId);
 
         $userContentInfo = $user->getVersionInfo()->getContentInfo();
-        $userMainLocation = $this->locationService->loadLocation($userContentInfo->mainLocationId);
         $contentType = $this->contentTypeService->loadContentType($userContentInfo->contentTypeId);
 
+        $mainLocationId = $userContentInfo->mainLocationId;
+
+        // special case for loading current user, which does not have permissions to load its location.
+        if ($user->id == $this->repository->getCurrentUser()->id) {
+            $locationService = $this->locationService;
```

Comment:
This is not needed any more as we require PHP 5.4 as of eZ 5.4, unless we want to backport this to 5.3.